src/zfinx.tex

```diff
@@ -0,0 +1,134 @@
+\mbox{{\em w}-1:0} of an {\tt x} register.
+Floating-point operations on {\em w}-bit operands ignore operand bits
+\mbox{XLEN-1:{\em w}}.
+
+Floating-point operations that produce \mbox{{\em w} $<$ XLEN-bit} results
+fill bits \mbox{XLEN-1:{\em w}} of the result with ones.
+
+\begin{commentary}
+To avoid the need for dedicated floating-point load instructions that fill the
+MSBs of an {\tt x} register with ones, we abandon the usual NaN-boxing
+requirement for floating-point operands.
+We retain the NaN-boxing of results to keep Zfinx as similar as possible and
+to ease debugging.
+\end{commentary}
+
+\section{Zhinx}
+
+The Zhinx extension provides analogous half-precision floating-point
+instructions.
+The Zhinx extension requires the Zfinx extension.
+
+The Zhinx extension adds all of the instructions that the Zfh extension
+adds, {\em except} for the transfer instructions FLH, FSH, FMV.H.X,
+and FMV.X.H.
+
+The Zhinx variants of these Zfh-extension instructions have the same semantics,
+except that whenever such an instruction would have accessed an {\tt f}
+register, it instead accesses the {\tt x} register with the same number.
+
+\section{Zdinx}
+
+The Zdinx extension provides analogous double-precision floating-point
+instructions.
+The Zdinx extension requires the Zfinx extension.
+
+The Zdinx extension adds all of the instructions that the D extension
+adds, {\em except} for the transfer instructions FLD, FSD, FMV.D.X,
+FMV.X.D, C.FLD[SP], and C.FSD[SP].
+
+The Zdinx variants of these D-extension instructions have the same semantics,
+except that whenever such an instruction would have accessed an {\tt f}
+register, it instead accesses the {\tt x} register with the same number.
+
+\section{Processing of Wider Values}
+
+Double-precision operands in RV32Zdinx
+are held in aligned {\tt x}-register pairs, i.e.,
+register numbers must be even.
+Use of misaligned (odd-numbered) registers for double-width floating-point
+operands is {\em reserved}.
+
+Regardless of endianness, the lower-numbered register holds the low-order
+bits, and the higher-numbered register holds the high-order bits: e.g., bits
+31:0 of a double-precision operand in RV32Zdinx might be held in register
+{\tt x14}, with bits 63:32 of that operand held in {\tt x15}.
+
+When a double-width floating-point result is written to {\tt x0}, the entire
+write takes no effect: e.g., for RV32Zdinx, writing a double-precision result
+to {\tt x0} does not cause {\tt x1} to be written.
+
+When {\tt x0} is used as a double-width floating-point operand, the entire
+operand is zero---i.e., {\tt x1} is not accessed.
+
+\begin{commentary}
+Load-pair and store-pair instructions are not provided, so transferring
+double-precision operands in RV32Zdinx from or to memory requires
+two loads or stores.
+Register moves need only a single FSGNJ.D instruction, however.
+\end{commentary}
+
+\begin{commentary}
+In future, an RV64Zqinx quad-precision extension could be defined analogously
```

Comment:
```suggestion
In the future, an RV64Zqinx quad-precision extension could be defined analogously
```

Reply:
I think the former is a correct Britishism, but yeah, don't see it in American English much.